modules/ding_gatewayf/ding_gatewayf.module

```diff
@@ -255,12 +283,66 @@ function ding_gatewayf_ding_provider_user() {
 }
 
 /**
+ * Do a check on the checksum given from gatewayf.
+ *
+ * @param array $attributes
+ *   The attributes from gatewayf.
+ *
+ * @return bool
+ *   Safe or not.
+ */
+function ding_gatewayf_security_check(array $attributes) {
+  // Check attributes.
+  if (empty($attributes['schacPersonalUniqueID']) || empty($attributes['eduPersonTargetedID'])) {
+    // We expect alway to get these two attributes.
+    watchdog('ding_gatewayf', 'Neccesary attributes missing', array(), WATCHDOG_ERROR);
+    drupal_set_message('Sorry login failed - contact site administrator', 'warning');
+    return FALSE;
+  }
```

Comment:
I would still think this should be removed. The current implementation will set an error message and return `FALSE`. The calling code checks for a non-`TRUE` response and set the exact same error message.